### PR TITLE
Introduce rabbitmq-plugins is_enabled [plugin 1] [plugin 2] [...]

### PR DESCRIPTION
**Pull request by michaelklishin**
Part of #292.

To QA:

1. Start a node from `umbrella/deps/rabbitmq_server_release` the same was as for running the test suite: `gmake run-broker PLUGINS="rabbitmq_federation rabbitmq_stomp"`

2. Switch to `deps/rabbit`, run `gmake`

3.  Try various combinations

``` bash
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp --offline --formatter=json; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp rabbitmq_local --offline; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp rabbitmq_local; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_local; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp --offline; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp rabbitmq_federation --offline; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp rabbitmq_federation; echo $?;
./scripts/rabbitmq-plugins -q is_enabled rabbitmq_stomp rabbitmq_local; echo $?
```

### CODE REVIEW

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  def plugin_or_plugins(args) when length(args) == 1 do
+    "plugin #{PluginHelpers.comma_separated_names(args)} is"
+  end
+  def plugin_or_plugins(args) when length(args) > 1 do
+    "plugins #{PluginHelpers.comma_separated_names(args)} are"
+  end
+
+  defp positive_result_message(args, %{online: true, node: node_name}) do
+    String.capitalize "#{plugin_or_plugins(args)} enabled on node #{node_name}"
```
[the remark this answers is not in the document]
Comment (gerhard) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: This is a nice feature:

```
rabbitmq-plugins is_enableds rabbitmq_stomp

Command 'is_enableds' not found.
Did you mean 'is_enabled'?
```

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      plugins  ->
+        plugins = Enum.map(plugins, &Atom.to_string/1) |> Enum.sort
+        case Enum.reduce(args, true,
+              fn p, acc -> acc && Enum.member?(plugins, p) end) do
+          true  -> {:ok,    positive_result_message(args, opts)}
+          false -> {:error, negative_result_message(args, opts, plugins)}
+        end
+    end
+  end
+  def run(args, %{offline: true} = opts) do
```
[the remark this answers is not in the document]
Comment (michaelklishin) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: I think it's up to the user to provide valid plugin names. There are no requirements as to what plugin names should be like in general.

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+  defp positive_result_message(args, %{online: true, node: node_name}) do
+    String.capitalize "#{plugin_or_plugins(args)} enabled on node #{node_name}"
+  end
+  defp positive_result_message(args, %{offline: true}) do
+    String.capitalize "#{plugin_or_plugins(args)} enabled"
+  end
+
+  defp negative_result_message(args, %{online: true, node: node_name}, plugins) do
+    String.capitalize("#{plugin_or_plugins(args)} not (all) enabled on node #{node_name}. ")
```
[the remark this answers is not in the document]
Comment (michaelklishin) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: Some may be enabled but not all of them, which is what the command expects when more than one plugin is listed.

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelklishin) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: I don't mind adding parentheses.

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (gerhard) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: I would agree in the case of invalid names, but typos are too easy to make. Happy to leave it out of scope for v1.

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex (hunk 3 above)
[the remark this answers is not in the document]
Comment (gerhard) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: In which case, should we split the message in available & unavailable? I would have never guessed the meaning from the initial output message.

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (michaelklishin) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: I don't think we should be building any heuristics into such a basic command. We do list what plugins are enabled (and so does `rabbitmqctl list [--enabled]`).

File: lib/rabbitmq/cli/plugins/commands/is_enabled.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaelklishin) [reply lib/rabbitmq/cli/plugins/commands/is_enabled.ex]: I cannot reproduce any string formatted as binaries.

**Pull request merged**